internal/tracer/tracer.go

```diff
@@ -125,33 +176,42 @@ func initTracer(serviceName string, c conftypes.WatchableSiteConfig) {
 
 		oldOpts = opts
 
-		tracer, closer, err := newTracer(&opts)
+		t, closer, err := newTracer(&opts)
 		if err != nil {
-			log15.Warn("Could not initialize jaeger tracer", "error", err.Error())
+			log15.Warn("Could not initialize tracer", "tracer", opts.tracerType, "error", err.Error())
 			return
 		}
-
-		globalTracer.set(tracer, closer, opts.Debug)
+		globalTracer.set(t, closer, opts.debug)
 	})
 }
 
-func newTracer(opts *jaegerOpts) (opentracing.Tracer, io.Closer, error) {
-	if !opts.Enabled {
-		log15.Info("opentracing: Jaeger disabled")
+// TODO Use openTelemetry https://github.com/sourcegraph/sourcegraph/issues/27386
+func newTracer(opts *options) (opentracing.Tracer, io.Closer, error) {
+	if opts.tracerType == None {
+		log15.Info("tracing disabled")
+		ddtracer.Stop()
 		return opentracing.NoopTracer{}, nil, nil
 	}
+	if opts.tracerType == Datadog {
+		log15.Info("Datadog: tracing enabled")
+		tracer := ddopentracing.New(ddtracer.WithService(opts.serviceName),
+			ddtracer.WithDebugMode(opts.debug),
+			ddtracer.WithServiceVersion(opts.version), ddtracer.WithEnv(opts.env))
+		return tracer, nil, nil
+	}
+	ddtracer.Stop()
 
-	log15.Info("opentracing: Jaeger enabled")
+	log15.Info("opentracing:  enabled")
 	cfg, err := jaegercfg.FromEnv()
-	cfg.ServiceName = opts.ServiceName
+	cfg.ServiceName = opts.serviceName
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "jaegercfg.FromEnv failed")
 	}
 	cfg.Tags = append(cfg.Tags, opentracing.Tag{Key: "service.version", Value: version.Version()})
 	if reflect.DeepEqual(cfg.Sampler, &jaegercfg.SamplerConfig{}) {
 		// Default sampler configuration for when it is not specified via
 		// JAEGER_SAMPLER_* env vars. In most cases, this is sufficient
-		// enough to connect Sourcegraph to Jaeger without any env vars.
+		// enough to connect Sourcegraph to  without any env vars.
```

Comment:
```suggestion
		// enough to connect Sourcegraph to Jaeger without any env vars.
```

I believe this is still relevant to Jaeger